src/components/uri/Visualization.tsx

```diff
@@ -387,6 +400,15 @@ export class VisualizationWrapped
             });
         this.subject.next(promise);
     }
+
+    private async getColorPalette() {
+        if (this.state.colorPaletteEnabled) {
```

Comment:
Maybe we should also skip loading of palette if given directly in props

Reply:
Fixed